http/analysis-service/server.py

```diff
@@ -846,6 +849,135 @@ def kill(instance_id):
+                                                's3://telemetry-spark-emr/telemetry.sh',
+                                                ['--public-key', pubkey])
+
+    jobflow_id = emr.run_jobflow(name = request.form['token'],
+                              ec2_keyname = 'mozilla_vitillo',
+                              master_instance_type = app.config['MASTER_INSTANCE_TYPE'],
+                              slave_instance_type = app.config['SLAVE_INSTANCE_TYPE'],
+                              num_instances = n_instances,
+                              ami_version = app.config['AMI_VERSION'],
+                              service_role = 'EMR_DefaultRole',
+                              job_flow_role = 'telemetry-spark-emr',
+                              visible_to_all_users = True,
+                              keep_alive = True,
+                              bootstrap_actions = [install_spark_bootstrap, setup_telemetry_bootstrap])
+
+    # Associate a few tags
+    emr.add_tags(jobflow_id, {
+        "Owner": current_user.email,
+        "Name": request.form['name'],
+        "Application": app.config['INSTANCE_APP_TAG']
+    })
+
+    # Send an email to the user who launched it
+    params = {
+        'monitoring_url': abs_url_for('cluster_monitor', jobflow_id = jobflow_id)
+    }
+    ses.send_email(
+        source = app.config['EMAIL_SOURCE'],
+        subject = ("telemetry-analysis cluster: %s (%s) launched" % (request.form['name'], jobflow_id)),
+        format = 'html',
+        body = render_template('cluster/email.html', **params),
+        to_addresses = [current_user.email]
+    )
+
+    return redirect(url_for('cluster_monitor', jobflow_id = jobflow_id))
+
+@app.route("/cluster/monitor/<jobflow_id>", methods=["GET"])
+@login_required
+def cluster_monitor(jobflow_id):
+    # Check that the user logged in is also authorized to do this
+    if not current_user.is_authorized():
+        return login_manager.unauthorized()
+
+    try:
+        jobflow = emr.describe_jobflow(jobflow_id)
+    except:
+        return "No such cluster: {}".format(jobflow_id), 404
+
+    # Alright then, let's report status
+    return render_template(
+        'cluster/monitor.html',
+        jobflow_id = jobflow_id,
+        instance_state = jobflow.state,
+        public_dns = jobflow.masterpublicdnsname if hasattr(jobflow, "masterpublicdnsname") else None,
+        terminate_url = abs_url_for('cluster_kill', jobflow_id = jobflow_id)
+    )
+
+@app.route("/cluster/kill/<jobflow_id>", methods=["GET"])
+@login_required
+def cluster_kill(jobflow_id):
+    # Check that the user logged in is also authorized to do this
+    if not current_user.is_authorized():
+        return login_manager.unauthorized()
+
+    try:
+        jobflow = emr.describe_jobflow(jobflow_id)
+    except:
+        return "No such cluster: {}".format(jobflow_id), 404
+
+    # Terminate cluster
+    emr.terminate_jobflow(jobflow_id)
```

Comment:
Users should **definitely** only be able to kill their own clusters :) Check the `Owner` tag to make sure it matches the logged in user's email.